# Proyek Analisis Data: [Input Nama Dataset]
- **Nama:** 
- **Email:** 
- **ID Dicoding:** [Input Username]

## Menentukan Pertanyaan Bisnis

- Pertanyaan 1 -> 
- Pertanyaan 2

## Import Semua Packages/Library yang Digunakan

In [8]:
import pandas as pd 

## Data Wrangling

### Gathering Data

In [9]:
#customer dataset
customers_df=pd.read_csv('data/customers_dataset.csv')
geolocation_df=pd.read_csv('data/geolocation_dataset.csv')
order_items_df=pd.read_csv('data/order_items_dataset.csv')
order_payments_df=pd.read_csv('data/order_payments_dataset.csv')
order_reviews_df=pd.read_csv('data/order_reviews_dataset.csv')
orders_df=pd.read_csv('data/orders_dataset.csv')
products_category_df=pd.read_csv('data/product_category_name_translation.csv')
products_df=pd.read_csv('data/products_dataset.csv')
sellers_df=pd.read_csv('data/sellers_dataset.csv')

In [10]:
products_df = products_df.merge(products_category_df, on='product_category_name', how='left')

In [11]:
df=orders_df.merge(order_items_df, on='order_id', how='inner')
df=df.merge(order_payments_df, on='order_id', how='inner',validate='m:m')
df=df.merge(order_reviews_df, on='order_id', how='inner')
df=df.merge(products_df, on='product_id',how='inner' )
df=df.merge(customers_df, on='customer_id', how='inner')
df=df.merge(sellers_df, on='seller_id',how='inner')

In [12]:
df.head()

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
2  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
3  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
4  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
2    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
3    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
4    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2017-10-04 19:55:00           2017-10-10 21:25:13   
2          2017-10-04 19:55:00           2017-10-10 21:25:13   
3          2018-07-26 14:31:00           2018-08-07 15:27:45   
4          2018-08-08 13:50:00           2018-08-17 18:06:29   

  order_estimated_delivery_date  order_item_id  \
0           2017-10-18 00:00:00              1   
1           2017-10-18 00:00:00              1   
2           2017-10-18 00:00:00              1   
3           2018-08-13 00:00:00              1   
4           2018-09-04 00:00:00              1   

                         product_id  ... product_height_cm product_width_cm  \
0  87285b34884572647811a353c7ac498a  ...               8.0             13.0   
1  87285b34884572647811a353c7ac498a  ...               8.0             13.0   
2  87285b34884572647811a353c7ac498a  ...               8.0             13.0   
3  595fac2a385ac33a80bd5114aec74eb8  ...              13.0             19.0   
4  aa4383b373c6aca5d8797843e5594415  ...              19.0             21.0   

   product_category_name_english                customer_unique_id  \
0                     housewares  7c396fd4830fd04220f754e42b4e5bff   
1                     housewares  7c396fd4830fd04220f754e42b4e5bff   
2                     housewares  7c396fd4830fd04220f754e42b4e5bff   
3                      perfumery  af07308b275d755c9edb36a90c618231   
4                           auto  3a653a41f6f9fc3d2a113cf8398680e8   

   customer_zip_code_prefix customer_city  customer_state  \
0                      3149     sao paulo              SP   
1                      3149     sao paulo              SP   
2                      3149     sao paulo              SP   
3                     47813     barreiras              BA   
4                     75265    vianopolis              GO   

   seller_zip_code_prefix     seller_city  seller_state  
0                    9350            maua            SP  
1                    9350            maua            SP  
2                    9350            maua            SP  
3                   31570  belo horizonte            SP  
4                   14840         guariba            SP  

[5 rows x 40 columns]

**Insight:**
- Total 39 columns dan 117329 baris 
- data berhasil digabungkan

### Assessing Data

In [13]:
df.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117329 entries, 0 to 117328
Data columns (total 40 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   order_id                       117329 non-null  object 
 1   customer_id                    117329 non-null  object 
 2   order_status                   117329 non-null  object 
 3   order_purchase_timestamp       117329 non-null  object 
 4   order_approved_at              117314 non-null  object 
 5   order_delivered_carrier_date   116094 non-null  object 
 6   order_delivered_customer_date  114858 non-null  object 
 7   order_estimated_delivery_date  117329 non-null  object 
 8   order_item_id                  117329 non-null  int64  
 9   product_id                     117329 non-null  object 
 10  seller_id                      117329 non-null  object 
 11  shipping_limit_date            117329 non-null  object 
 12  price                         

In [ ]:
df.isna().sum()

order_id                              0
customer_id                           0
order_status                          0
order_purchase_timestamp              0
order_approved_at                    15
order_delivered_carrier_date       1235
order_delivered_customer_date      2471
order_estimated_delivery_date         0
order_item_id                         0
product_id                            0
seller_id                             0
shipping_limit_date                   0
price                                 0
freight_value                         0
payment_sequential                    0
payment_type                          0
payment_installments                  0
payment_value                         0
review_id                             0
review_score                          0
review_comment_title             103437
review_comment_message            67650
review_creation_date                  0
review_answer_timestamp               0
product_category_name              1695


**Insight:**

Missing value pada:
- order_approved_at -> 15
- order_delivered_carrier_date -> 1235
- order_delivered_customer_date -> 2471
- review_comment_title -> 103437
- review_comment_message -> 67650
- product_category_name -> 1695


-  order_items_df -> terdapat kesalahan tipe data untuk kolom shipping_limit_date
- order_reviews_df -> Terdapat kesalahan tipe data untuk kolom review_creation_date dan review_answer_timestamp. Terdapat 87656 missing value pada kolom review_comment_title, 58247 missing value pada kolom review_comment_message.
- orders_df -> terdapat kesalahan tipe data untuk kolom order_purchase_timestamp, order_approved_at, order_delivered_carrier_date, order_delivered_customer_date, order_estimated_delivery_date. Terdapat 160 missing value pada kolom order_approved_at, 1783 missing value pada kolom order_delivered_carrier_date, 2965 missing value pada kolom order_delivered_customer_date. 
- products_df -> terdapat 160 missing value pada kolom product_category_name,
product_name_lenght, product_description_lenght, product_photos_qty. terdapat 2 missing value pada kolom product_weight_g, product_length_cm, product_height_cm, product_width_cm.

### Cleaning Data

#### geolocation

In [29]:
# handle duplicate geolocation
geolocation_df.drop_duplicates(inplace=True)

#### order items

In [30]:
order_items_df['shipping_limit_date']=pd.to_datetime(order_items_df['shipping_limit_date'])
order_items_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype         
---  ------               --------------   -----         
 0   order_id             112650 non-null  object        
 1   order_item_id        112650 non-null  int64         
 2   product_id           112650 non-null  object        
 3   seller_id            112650 non-null  object        
 4   shipping_limit_date  112650 non-null  datetime64[ns]
 5   price                112650 non-null  float64       
 6   freight_value        112650 non-null  float64       
dtypes: datetime64[ns](1), float64(2), int64(1), object(3)
memory usage: 6.0+ MB


#### order reviews

In [31]:
datetime_coloms=['review_creation_date','review_answer_timestamp']
for column in datetime_coloms:
    order_reviews_df[column]=pd.to_datetime(order_reviews_df[column])

In [32]:
order_reviews_df.fillna(value='no coment', inplace=True)

#### order

In [35]:
orders_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  object
 1   customer_id                    99441 non-null  object
 2   order_status                   99441 non-null  object
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(8)
memory usage: 6.1+ MB


**Insight:**
- xxx
- xxx

## Exploratory Data Analysis (EDA)

### Explore ...

**Insight:**
- xxx
- xxx

## Visualization & Explanatory Analysis

### Pertanyaan 1:

### Pertanyaan 2:

**Insight:**
- xxx
- xxx

## Analisis Lanjutan (Opsional)

## Conclusion

- Conclution pertanyaan 1
- Conclution pertanyaan 2